snudda/utils/swap_to_degenerated_morphologies_extended.py

```diff
@@ -165,27 +221,39 @@ def filter_synapses(self, filter_axon=False):
 
         # Keep synapses in original network that are still within the new morphologies
         for synapses in self.synapse_iterator():
+
+            # if synapses[0, 0] == 4 and synapses[0, 1] == 100:
+            #     print("tell me why...")
+            #     import pdb
+            #     pdb.set_trace()
+
             new_syn = self.filter_synapses_helper(synapses, filter_axon=filter_axon)                    
             new_synapses[syn_ctr:syn_ctr + new_syn.shape[0]] = new_syn
             syn_ctr += new_syn.shape[0]
 
         # Here add the new synapses from growing axons and dendrites
-        additional_synapses = self.get_additional_synapses()
+        additional_synapses = self.get_additional_synapses(network_config=config, rng=self.rng)
         new_synapses[syn_ctr:syn_ctr+additional_synapses.shape[0], :] = additional_synapses
         syn_ctr += additional_synapses.shape[0]
 
         # Resort the new synapse matrix
         sorted_synapses = self.sort_synapses(new_synapses[:syn_ctr, :])
 
         old_synapse_iterator = self.synapse_iterator(synapses=self.old_hdf5["network/synapses"][()])
-        config = json.loads(self.updated_network_loader.data["config"], object_pairs_hook=OrderedDict)
 
-        pruned_synapses = self.post_degeneration_pruning(synapses=sorted_synapses,
-                                                         old_synapse_iterator=old_synapse_iterator,
-                                                         network_config=config,
-                                                         rng=self.rng)
```

Comment:
I suggest to remove the if-else statement here

Reply:
Is this comment supposed to refer to another line?

Reply:
found it